katsdptelstate/rdb_writer.py

```diff
@@ -16,145 +16,146 @@
 
 import logging
 import os
+from contextlib import contextmanager
 
-from .endpoint import endpoint_parser
 from .rdb_utility import encode_len
+from .telescope_state import _ensure_binary, _display_str
 
-
+# Basic RDB header. First 5 bytes are the standard REDIS magic
+# Next 4 bytes store the RDB format version number (6 in this case)
+# 0xFE flags the start of the DB selector (which is set to 0)
 RDB_HEADER = b'REDIS0006\xfe\x00'
- # Basic RDB header. First 5 bytes are the standard REDIS magic
- # Next 4 bytes store the RDB format version number (6 in this case)
- # 0xFE flags the start of the DB selector (which is set to 0)
 
 RDB_TERMINATOR = b'\xFF'
 
+# Fast CRC-64 implementations on Python seem rare
+# RDB is happy with zero checksum - we will protect the RDB dump in other ways
 RDB_CHECKSUM = b'\x00\x00\x00\x00\x00\x00\x00\x00'
- # Fast CRC-64 implementations on Python seem rare
- # RDB is happy with zero checksum - we will protect the RDB
- # dump in other ways
 
-DEFAULT_PORT = 6379
- # Default Redis port for use in endpoint constructors
+logger = logging.getLogger(__name__)
 
 
-class RDBWriter(object):
-    """Very limited RDB dump utility used to dump a specified subset
-    of keys from an active Redis DB to a valid RDB format file.
+def encode_item(key, key_dump):
+    """Encode key and corresponding DUMPed value to RDB format.
 
-    Either an endpoint, a compatible redis client, or a
-    :class:`katsdptelstate.Backend` must be provided.
+    First byte is used to indicate the encoding used for the value of this key.
```

Comment:
"First byte" -> "The first byte"

Reply:
Done. Fixed up the whole docstring while I'm at it.

Reply:
Actually, @sratcliffe, the `encode_item` docstring talks about 1-byte and 2-byte lengths, but the actually used `encode_len` also does 5-byte lengths. Is this docstring still relevant?

Reply:
Probably too much detail since it is described in encode_len - may as well remove.

Reply:
Removed.